backend/service/github/github.go

```diff
@@ -268,11 +268,15 @@ type CreateBranchRequest struct {
 
 	// The commit message for files added.
 	CommitMessage string
+
+	// Fetch only ReferenceName specified branch.
+	SingleBranch bool
 }
 
 // Creates a new branch with a commit containing files and pushes it to the remote.
 func (s *svc) CreateBranch(ctx context.Context, req *CreateBranchRequest) error {
 	cloneOpts := &git.CloneOptions{
+		SingleBranch:  req.SingleBranch,
```

Comment:
Yes, this boolean field will have default value as false if it is not set in CreateBranchRequest, so it will not change the current behavior in those modules, I also tested it in TQ topic management module in clutch-private repo.